plugin/core/registry.py

```diff
@@ -101,11 +101,36 @@ def sessions(self, capability: Optional[str] = None) -> Generator[Session, None,
         yield from sessions_for_view(self.view, capability)
 
 
-class LspRestartClientCommand(sublime_plugin.TextCommand):
-    def run(self, edit: Any) -> None:
+class LspRestartServerCommand(LspTextCommand):
+    def run(self, edit: Any, config_name: str = None, show_quick_panel: bool = False) -> None:
         window = self.view.window()
-        if window:
-            windows.lookup(window).restart_sessions_async()
+        if not window:
+            return
+        self._config_names = [session.config.name for session in self.sessions()] if not config_name else [config_name]
+        if not self._config_names:
+            return
+        self._wm = windows.lookup(window)
+        if len(self._config_names) == 1:
+            self.restart_server(0)
+        else:
+            self._config_names.insert(0, 'All Servers')
+            if not show_quick_panel:
+                self.restart_server(0)
+            else:
+                window.show_quick_panel(self._config_names, self.restart_server)
+
+    def restart_server(self, index: int) -> None:
+        if index > -1:
+
+            def run_async() -> None:
+                config_name = self._config_names[index]
+                if not config_name or not self.session_by_name(config_name):
```

Comment:
It would be nice to be more explicit here and instead of restarting all sessions when config_name doesn't match any session, check for "All Servers" specifically. There could be a class constant for "All Server" to make it DRY.